# 4.9.3. Relaciones entre los esfuerzos principales obtenidos en el análisis bidimensional y tridimensional

Por:
- Michael Heredia Pérez
- [mherediap@unal.edu.co](mherediap@unal.edu.co)
- 2026-02-18





## **Recordemos el Ejemplo 2.6 - Esfuerzos y direcciones principales**

Consideramos el estado plano de esfuerzos:

$$
\sigma_x = -1\ \text{Pa}, \qquad
\sigma_y = 2\ \text{Pa}, \qquad
\tau_{xy} = -3\ \text{Pa}
$$

El tensor de tensiones es:

$$
\boldsymbol{\sigma}=
\begin{bmatrix}
\sigma_x  & \tau_{xy} \\
\tau_{xy} & \sigma_y
\end{bmatrix}
$$


In [1]:
import numpy as np

# Datos
sx = -1.0
sy = 2.0
txy = -3.0

sigma = np.array([[sx, txy],
                  [txy, sy]])

sigma

array([[-1., -3.],
       [-3.,  2.]])

### Esfuerzos principales

Los esfuerzos principales se obtienen como los autovalores del tensor:

$$
\det(\boldsymbol{\sigma}-\lambda \mathbf{I})=0
$$

La ecuación que aplicamos es:

$$
(\sigma_{2})_{xy} =
\frac{\sigma_x+\sigma_y}{2}
+
\sqrt{
\left(\frac{\sigma_x-\sigma_y}{2}\right)^2
+\tau_{xy}^2
}
$$

$$
(\sigma_{1})_{xy} =
\frac{\sigma_x+\sigma_y}{2}
-
\sqrt{
\left(\frac{\sigma_x-\sigma_y}{2}\right)^2
+\tau_{xy}^2
}
$$

In [2]:
# Calculamos cada componentes de la ecuación.
sigma_avg = (sx + sy)/2
R = np.sqrt(((sx - sy)/2)**2 + txy**2)

# Calculamos los esfuerzos principales.
sigma_1 = sigma_avg + R
sigma_2 = sigma_avg - R

# Presentamos los resultados.
print(f"sigma_1 = {round(sigma_1, 4)}")
print(f"sigma_2 = {round(sigma_2, 4)}")

sigma_1 = 3.8541
sigma_2 = -2.8541


### 2. Dirección principal

El ángulo asociado al esfuerzo principal satisface:

$$
\tan(2\theta_1)=\frac{ + \tau_{xy}}{+\frac{\sigma_x-\sigma_y}{2}}
$$

Para evitar ambigüedad de cuadrante usamos la función ```atan2```:

$$
\theta_1 = \frac{1}{2},\text{atan2}(2\tau_{xy},\sigma_x-\sigma_y)
$$

In [3]:
# Calculamos la dirección principal asociada a sigma_1.
theta_1 = 0.5*np.arctan2(2*txy, sx - sy)
theta_1_deg = np.degrees(theta_1)

# Presentamos resultados.
print(f"theta_1 = {round(theta_1, 4)} rad")
print(f"theta_1 = {round(theta_1_deg, 2)}°")

theta_1 = -1.0172 rad
theta_1 = -58.28°



El segundo plano principal está a:

$$
\theta_2=\theta_1+\frac{\pi}{2}
$$


In [4]:
# Calculamos sobre el valor en radianes.
theta_2 = theta_1 + np.pi/2
theta_2_deg = np.degrees(theta_2)

# Presentamos resultados.
print(f"theta_2 = {round(theta_2, 4)} rad")
print(f"theta_2 = {round(theta_2_deg, 2)}°")


theta_2 = 0.5536 rad
theta_2 = 31.72°


**Nota**: Verificar la graficación de estas direcciones en la Figura **2.20** y el círculo de Mohr asociado en la Figure **2.21**.

### 3. Esfuerzo cortante máximo

La magnitud del esfuerzo cortante máximo es el radio del círculo de Mohr:

$$
\tau_{max}=
\sqrt{
\left(\frac{\sigma_x-\sigma_y}{2}\right)^2
+\tau_{xy}^2
}
$$

Y ocurre a planos ubicados a:

$$
\theta_{c1}=\theta_1 -  45^\circ
$$

$$
\theta_{c2}=\theta_1 +  45^\circ
$$

In [5]:
# Ya tenemos una expresión para el tau_max.
tau_max = R
tau_min = -R
print(f"tau_max = {round(tau_max, 4)} Pa")
print(f"tau_min = {round(tau_min, 4)} Pa")

# Calculamos las direcciones asociadas.
theta_c1 = theta_1_deg - 45
theta_c2 = theta_1_deg + 45

# Presentamos resultados.
print()
print(f"theta_c1 = {round(theta_c1, 2)}°")
print(f"theta_c2 = {round(theta_c2, 2)}°")

tau_max = 3.3541 Pa
tau_min = -3.3541 Pa

theta_c1 = -103.28°
theta_c2 = -13.28°


### 4. Verificación mediante autovalores

Recordemos que los esfuerzos principales son los autovalores del tensor:

$$
\boldsymbol{\sigma}\boldsymbol{\hat{n}}=\sigma_n\boldsymbol{\hat{n}}
$$

In [6]:
# Resolvemos el problema de valores y vectores propios
vals, vecs = np.linalg.eigh(sigma)

print(f"Los valores propios son: \n{vals}\n")
print(f"Los vectores propios son: \n{vecs}\n")

Los valores propios son: 
[-2.85410197  3.85410197]

Los vectores propios son: 
[[-0.85065081 -0.52573111]
 [-0.52573111  0.85065081]]



**Nota** Recuerde que cada columna de ```vecs``` es un vector propio.

Observe que:

* Los autovalores coinciden con $\sigma_1$ y $\sigma_2$.
* Los autovectores coinciden con:

$$
\mathbf{n}_1=
\begin{bmatrix}
\cos\theta_1 \
\cos\theta_1
\end{bmatrix}
$$


In [7]:
print(f"cos(theta_1) = {np.cos(theta_1)}")
print(f"cos(theta_2) = {np.cos(theta_2)}")

cos(theta_1) = 0.5257311121191336
cos(theta_2) = 0.85065080835204


Recordemos que el signo del vector propio no es relevante; únicamente su dirección.

## **Comparación entre el análisis tridimensional y el bidimensional.**

En el ejemplo 2.6 resolvimos el problema en el plano $xy$ sin especificar si el estado correspondía a:

* Deformación plana
* Tensión plana

Ahora extenderemos el mismo estado de esfuerzos al caso tridimensional.

Recordemos el estado de esfuerzos:

$$
\sigma_x=-1\text{ Pa},\qquad
\sigma_y=2\text{ Pa},\qquad
\tau_{xy}=-3\text{ Pa}
$$

y que en 2D obtuvimos:

$$
(\sigma_1)*{xy}=3.8541\text{ Pa},\qquad
(\sigma_2)*{xy}=-2.8541\text{ Pa}
$$

In [8]:
# Aumentemos a 3D el problema inicial.
sigma_3D = np.array([[sx, txy, 0],
                  [txy, sy, 0],
                  [0, 0, 0]])

sigma_3D

array([[-1., -3.,  0.],
       [-3.,  2.,  0.],
       [ 0.,  0.,  0.]])

In [9]:
# Resolvemos el problema de valores y vectores propios
vals_3D, vecs_3D = np.linalg.eigh(sigma_3D)

print(f"Esfuerzos p.p. en 3D son: \n{vals_3D}\n")
print(f"Direcciones p.p. en 3D son: \n{vecs_3D}\n")

Esfuerzos p.p. en 3D son: 
[-2.85410197  0.          3.85410197]

Direcciones p.p. en 3D son: 
[[-0.85065081  0.         -0.52573111]
 [-0.52573111  0.          0.85065081]
 [-0.          1.          0.        ]]



### **Caso 1. Tensión plana**

Recordemos que en Tensión plana:
$$
\sigma_z=0,\qquad
\tau_{xz}=0,\qquad
\tau_{yz}=0
$$


In [10]:
sigma_tp = np.array([
    [sx, txy, 0],
    [txy, sy, 0],
    [0, 0, 0]
])

sigma_tp

array([[-1., -3.,  0.],
       [-3.,  2.,  0.],
       [ 0.,  0.,  0.]])

Resolvemos el problema de valores y vectores propios

In [11]:
vals_tp, vecs_tp = np.linalg.eigh(sigma_tp)

print(f"Esfuerzos p.p. en TP son: \n{vals_tp}\n")
print(f"Direcciones p.p. en TP: \n{vecs_tp}\n")

Esfuerzos p.p. en TP son: 
[-2.85410197  0.          3.85410197]

Direcciones p.p. en TP: 
[[-0.85065081  0.         -0.52573111]
 [-0.52573111  0.          0.85065081]
 [-0.          1.          0.        ]]



Aquí los valores propios son:

* Dos coinciden con $(\sigma_1)_{xy}$ y $(\sigma_2)_{xy}$.
* El tercero es cero.

Ordenando:

$$
\sigma_1=3.8541\text{ Pa}
$$
$$
\sigma_2=0\text{ Pa}
$$
$$
\sigma_3=-2.8541\text{ Pa}
$$


### **Caso 2. Deformación Plana**

Recordemos que en deformación plana:

* $\tau_{xz}=0$
* $\tau_{yz}=0$
* $\sigma_z=\nu(\sigma_x+\sigma_y)$

Por tanto:

$$
\boldsymbol{\sigma}=
\begin{bmatrix}
\sigma_x & \tau_{xy} & 0 \\
\tau_{xy} & \sigma_y & 0 \\
0 & 0 & \nu(\sigma_x+\sigma_y)
\end{bmatrix}
$$

In [12]:
nu = 0.30

sigma_dp = np.array([
    [sx, txy, 0],
    [txy, sy, 0],
    [0, 0, nu*(sx+sy)]
])

sigma_dp

array([[-1. , -3. ,  0. ],
       [-3. ,  2. ,  0. ],
       [ 0. ,  0. ,  0.3]])

Resolvemos el problema de valores y vectores propios

In [13]:
vals_dp, vecs_dp = np.linalg.eig(sigma_dp)

print(f"Esfuerzos p.p. en DP son: \n{vals_dp}\n")
print(f"Direcciones p.p. en DP: \n{vecs_dp}\n")

Esfuerzos p.p. en DP son: 
[-2.85410197  3.85410197  0.3       ]

Direcciones p.p. en DP: 
[[-0.85065081  0.52573111  0.        ]
 [-0.52573111 -0.85065081  0.        ]
 [ 0.          0.          1.        ]]



**Observación**

* Dos valores propios coinciden con los obtenidos en 2D.
* El tercero es:

$$
\sigma_z=\nu(\sigma_x+\sigma_y)
$$

En este ejemplo:

$$
\sigma_z=0.3(-1+2)=0.3\text{ Pa}
$$

Luego, ordenando:

$$
\sigma_1=3.8541\text{ Pa}
$$
$$
\sigma_2=0.3\text{ Pa}
$$
$$
\sigma_3=-2.8541\text{ Pa}
$$

Observe que **el problema deja de ser puramente bidimensional**: ahora el segundo esfuerzo principal es el inducido por la restricción cinemática.


Observe la diferencia clave:

- En deformación plana:

$$
\sigma_2=\nu(\sigma_x+\sigma_y)
$$

- En tensión plana:

$$
\sigma_2=0
$$


In [14]:
# Calculamos el esfuerzo cortante máximo.
sigma_sorted = np.sort(vals_dp)[::-1]  # ordenar de mayor a menor
tau_max_dp = (sigma_sorted[0] - sigma_sorted[2]) / 2

print(f"tau_max_DP = {round(tau_max_dp, 4)} Pa")


tau_max_DP = 3.3541 Pa


Observe que:

$$
\tau_{max}=3.3541\text{ Pa}
$$

Coincide con el valor obtenido en 2D.

## **Conclusión**

En **Deformación plana** aplica:

$$
\sigma_1=\max\left((\sigma_1)*{xy},(\sigma_2)*{xy},\nu(\sigma_x+\sigma_y)\right)
$$

$$
\sigma_2=\text{mediana}\left((\sigma_1)*{xy},(\sigma_2)*{xy},\nu(\sigma_x+\sigma_y)\right)
$$

$$
\sigma_3=\min\left((\sigma_1)*{xy},(\sigma_2)*{xy},\nu(\sigma_x+\sigma_y)\right)
$$


En **Tensión plana** aplica:

$$
\sigma_1=\max\left((\sigma_1)*{xy},(\sigma_2)*{xy},0\right)
$$

$$
\sigma_2=\text{mediana}\left((\sigma_1)*{xy},(\sigma_2)*{xy},0\right)
$$

$$
\sigma_3=\min\left((\sigma_1)*{xy},(\sigma_2)*{xy},0\right)
$$